packages/plugin-conversation/src/share-activity.js

```diff
@@ -0,0 +1,234 @@
+      mimeType: file.type,
+      objectType: `file`,
+      [EMITTER_SYMBOL]: emitter
+    };
+
+    this.uploads.set(file, upload);
+
+    const promise = processImage(file, this.config.thumbnailMaxWidth, this.config.thumbnailMaxHeight, this.logger)
+      .then((imageData) => {
+        const main = this.spark.encryption.encryptBinary(file)
+          .then(({scr, cdata}) => {
+            upload.scr = scr;
+            return Promise.all([cdata, this.spaceUrl]);
+          })
+          .then(([cdata, spaceUrl]) => {
+            const uploadPromise = this._upload(cdata, `${spaceUrl}/upload_sessions`);
+            transferEvents(`progress`, uploadPromise, emitter);
+            return uploadPromise;
+          })
+          .then((metadata) => {
+            upload.url = upload.scr.loc = metadata.downloadUrl;
+          });
+
+
+        let thumb;
+        if (imageData) {
+          const [thumbnail, fileDimensions, thumbnailDimensions] = imageData;
+          Object.assign(upload, fileDimensions);
+
+          upload.image = thumbnailDimensions;
+          thumb = this.spark.encryption.encryptBinary(thumbnail)
+            .then(({scr, cdata}) => {
+              upload.image.scr = scr;
+              return Promise.all([cdata, this.hiddenSpaceUrl]);
+            })
+            .then(([cdata, spaceUrl]) => this._upload(cdata, `${spaceUrl}/upload_sessions`))
+            .then((metadata) => {
+              upload.image.url = upload.image.scr.loc = metadata.downloadUrl;
+            });
+        }
+
+        return Promise.all([main, thumb]);
+      });
+
+
+    upload[PROMISE_SYMBOL] = promise;
+
+    proxyEvents(emitter, promise);
+    return promise;
+  },
+
+  _upload(file, uri) {
+    const fileSize = file.length || file.size || file.byteLength;
+
+    return this.spark.upload({
+      uri,
+      file,
+      qs: {
+        transcode: true
+      },
+      phases: {
+        initialize: {fileSize},
+        upload: {
+          $url(session) {
+            return session.uploadUrl;
+          }
+        },
+        finalize: {
+          $uri(session) {
+            return session.finishUploadUrl;
+          },
+          body: {fileSize}
+        }
+      }
+    });
+  },
+
+  remove(file) {
+    this.uploads.delete(file);
+  },
+
+  prepare() {
+    const activity = {
+      verb: `share`,
+      object: {
+        objectType: `content`,
+        displayName: this.displayName,
+        content: this.content,
+        mentions: this.mentions,
+        files: {
+          items: []
+        }
+      },
+      clientTempId: this.clientTempId
+    };
+
+    const promises = [];
+    this.uploads.forEach((item) => {
+      activity.object.files.items.push(item);
+      promises.push(item[PROMISE_SYMBOL]);
+    });
+
+    activity.object.contentCategory = this._determineContentCategory(activity.object.files.items);
+
+    return Promise.all(promises)
+      .then(() => activity);
+  },
+
+  _determineContentCategory(items) {
```

Comment:
The contentCategory isn't asserted in the integration tests. Should we add it there to get coverage on this, or should we add unit tests to verify this is working correctly?